src/pipx/interpreter.py

```diff
@@ -58,6 +59,14 @@ def find_python_interpreter(python_version: str) -> str:
 
     if shutil.which(python_version):
         return python_version
+
+    if fetch_missing_python or FETCH_MISSING_PYTHON:
+        try:
+            standalone_executable = download_python_build_standalone(python_version)
+            return standalone_executable
+        except PipxError as e:
+            raise InterpreterResolutionError(source="PATH", version=python_version) from e
```

Comment:
I think we should add another source for `InterpreterResolutionError`?

Reply:
i suppose so... what should we call it?

Reply:
ok, i got it. it looks a bit verbose in code, but it should come out clear to the end user, and that's arguably more important.

check out [f241014](https://github.com/pypa/pipx/pull/1243/commits/f2410140622645c773b33b416d8e38eda6017e08) and tell me what you think

Reply:
Looks good! Resolving.